src/Planners/src/UnicyclePlanner.cpp

```diff
@@ -0,0 +1,707 @@
+    // ==================================
+
+    // Initialize the left FootPrint
+    m_pImpl->left = std::make_shared<FootPrint>();
+    m_pImpl->left->setFootName(m_pImpl->names.left);
+
+    // Initialize the right FootPrint
+    m_pImpl->right = std::make_shared<FootPrint>();
+    m_pImpl->right->setFootName(m_pImpl->names.right);
+
+    // Convert manif to iDynTree
+    auto toiDynTree = [](const manif::SE3d::Translation& translation) -> iDynTree::Vector2 {
+        iDynTree::Vector2 position;
+        position[0] = translation[0];
+        position[1] = translation[1];
+        return position;
+    };
+
+    // The initTime of the UnicyclePlanner cannot be smaller than
+    // the impact time of the last step.
+    // If an initial step configuration is passed, the initial time must be updated.
+    double initTime = m_pImpl->horizon.t0;
+
+    // Process the initial left contact configuration
+    if (m_pImpl->initialContacts.left)
+    {
+        const auto& contact = m_pImpl->initialContacts.left;
+
+        // Here we decompose the quaternion to YXZ intrinsic Euler angles (default in Eigen).
+        // The most reliable decomposition is ZXY extrinsic, that is equivalent since the two
+        // commute when the order is reversed. This decomposition, having Z as first rotation,
+        // should get the correct yaw in most cases.
+        const auto& euler
+            = contact->pose.quat().normalized().toRotationMatrix().eulerAngles(1, 0, 2);
+
+        // Create the inital step
+        m_pImpl->left->addStep(toiDynTree(contact->pose.translation()),
+                               euler[2],
+                               contact->activationTime);
+
+        const double impactTime = contact->activationTime;
+        initTime = impactTime > initTime ? impactTime : initTime;
+    }
+
+    // Process the initial left contact configuration
+    if (m_pImpl->initialContacts.right)
+    {
+        const auto& contact = m_pImpl->initialContacts.right;
+
+        // Here we decompose the quaternion to YXZ intrinsic Euler angles (default in Eigen).
+        // The most reliable decomposition is ZXY extrinsic, that is equivalent since the two
+        // commute when the order is reversed. This decomposition, having Z as first rotation,
+        // should get the correct yaw in most cases.
+        const auto& euler
+            = contact->pose.quat().normalized().toRotationMatrix().eulerAngles(1, 0, 2);
+
+        // Create the inital step
+        m_pImpl->right->addStep(toiDynTree(contact->pose.translation()),
+                                euler[2],
+                                contact->activationTime);
+
+        const double impactTime = contact->activationTime;
+        initTime = impactTime > initTime ? impactTime : initTime;
+    }
+
+    // Due to how the planner works, the init time must be bigger than the last activation
+    initTime = initTime > 0 ? initTime + 0.01 : initTime;
```

Comment:
I am a bit puzzled by this. Is it still needed?

Reply:
If  I recall, the origin of this workaround was to deal with the following section:

https://github.com/robotology/unicycle-footstep-planner/blob/d3f6c80afe21a9958da769c8dd8a2bbfee5ea922/src/UnicycleGenerator.cpp#L308-L312

Reply:
Nope, I deal with those lines in the section below, in the generator section. I think this can be removed now. The script I posted in the comment above seems working even without it.

Reply:
Actually, by reading this PR, I checked that code and I noticed that the error message is wrong, so thanks for that :grin: . Besides that, I added that check to make sure that there were no footsteps already planned in the future. In other words, it is pointless to say that the init time is 1, if at 2 there are already some planned footsteps. 


I thought you already dealt with that using https://github.com/ami-iit/bipedal-locomotion-framework/blob/a8fad53b2e0a9d46811479f2d51659105cc91923/src/Planners/src/UnicyclePlanner.cpp#L523. Is it still failing? :thinking:

Reply:
> Nope, I deal with those lines in the section below, in the generator section. I think this can be removed now. The script I posted in the comment above seems working even without it.

Nice!

Reply:
Yes in the beginning I was reading the error message and I was not understanding what was wrong with my logic. Then, I checked the logic of the unicycle planner, and realized that the error message was wrong :)